contracts/libraries/SafeMigration.sol

```diff
@@ -0,0 +1,136 @@
+        singleton = target;
+        emit ChangedMasterCopy(target);
+    }
+
+    /**
+     * @notice Migrate to Safe Singleton and set the fallback handler. This function is intended to be used when migrating
+     *         a Safe to a version which also requires updating fallback handler.
+     */
+    function migrateWithFallbackHandler(address target, address fallbackHandler) public onlyDelegateCall {
+        migrateSingleton(target);
+        require(getCodehash(fallbackHandler) == SAFE_FALLBACK_HANDLER_CODEHASH, "Invalid fallbackhandler");
+        ISafe(address(this)).setFallbackHandler(fallbackHandler);
+    }
+
+    /**
+     * @notice Migrate the Safe contract to a new Safe singleton implementation.
+     *         Checks whether the given target singleton address codehash matches the expected Safe/SafeL2 singleton codehash.
+     * @param target Address of the new Safe singleton implementation.
+     */
+    function migrateL2Singleton(address target) public onlyDelegateCall {
+        require(getCodehash(target) == SAFE_L2_SINGLETON_CODEHASH, "Invalid SafeL2 singleton");
+        singleton = target;
+        emit ChangedMasterCopy(target);
+    }
+
+    /**
+     * @notice Migrate to Safe Singleton (L2) and set the fallback handler. This function is intended to be used when migrating
+     *         a Safe to a version which also requires updating fallback handler.
+     */
+    function migrateL2WithFallbackHandler(address target, address fallbackHandler) public onlyDelegateCall {
+        migrateL2Singleton(target);
+        require(getCodehash(fallbackHandler) == SAFE_FALLBACK_HANDLER_CODEHASH, "Invalid fallbackhandler");
+
+        ISafe(address(this)).setFallbackHandler(fallbackHandler);
+    }
+
+    /**
+     * @notice Checks whether an Ethereum address corresponds to a contract or an externally owned account (EOA).
+     * @param account The Ethereum address to be checked.
+     * @return A boolean value indicating whether the address is associated with a contract (true) or an EOA (false).
+     * @dev This function relies on the `extcodesize` assembly opcode to determine whether an address is a contract.
+     * It may return incorrect results in some edge cases (see documentation for details).
+     * Developers should use caution when relying on the results of this function for critical decision-making.
+     */
+    function isContract(address account) internal view returns (bool) {
+        uint256 size;
+        /* solhint-disable no-inline-assembly */
+        /// @solidity memory-safe-assembly
+        assembly {
+            size := extcodesize(account)
+        }
+        /* solhint-enable no-inline-assembly */
+
+        // If the code size is greater than 0, it is a contract; otherwise, it is an EOA.
+        return size > 0;
+    }
+
+    /**
+     * @notice Get the codehash of an account.
+     * @param account The address of the account to get the codehash of
```

Comment:
missing `@return codeHash The code hash of the account.

Reply:
removed this function as it is not needed altogether